File: dispositivi-service/src/main/java/com/example/dispositivi/model/Totem.java
package com.example.dispositivi.model;

import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;

@Entity
@DiscriminatorValue("TOTEM")
public class Totem extends Dispositivo {

    public Totem() {
        super();
    }

    public Totem(Boolean status, Integer corsia, Integer casello) {
        super(status, corsia, casello);
    }

    public Totem(Integer ID, Boolean status, Integer corsia, Integer casello) {
        super(ID, status, corsia, casello);
    }
}

File: IOT/Totem.py
class Totem(Dispositivo): # Ereditiamo da Dispositivo
    def __init__(self, schermo, ir, corsia, casello, dispositivo_id=None):
        super().__init__(
            status=True,
            corsia=corsia,
            casello=casello,
            tipo_dispositivo="TOTEM",
            dispositivo_id=dispositivo_id
        )
        self.schermo = schermo
        self.ir = ir